x-pack/plugins/infra/public/alerting/metric_threshold/components/custom_equation/metric_row_with_count.tsx

```diff
@@ -89,7 +86,14 @@ export const MetricRowWithCount = ({
               { defaultMessage: 'KQL Filter {name}', values: { name } }
             )}
           >
-            <EuiFieldText compressed value={filter} onChange={handleFilterChange} />
+            <MetricsExplorerKueryBar
```

Comment:
Ah interesting. I built a Query builder component for the SLO Edit Form here: `x-pack/plugins/observability/public/pages/slo_edit/components/custom_kql/query_builder.tsx`. That one is wrapped in some stuff to make it useable in the context of the SLO form, but when you would remove the wrapping components, it's just:

```
import React from 'react';
import { QueryStringInput } from '@kbn/unified-search-plugin/public';
import { useKibana } from '../../../../utils/kibana_react';
import { useCreateDataView } from '../../../../hooks/use_create_data_view';

export interface Props {
  dataTestSubj: string;
  indexPatternString: string | undefined;
  query: string;
  placeholder: string;
  onChange: (value: string) => void;
}

export function QueryBuilder({
  dataTestSubj,
  indexPatternString,
  query,
  placeholder,
  onChange,
}: Props) {
  const { data, dataViews, docLinks, http, notifications, storage, uiSettings, unifiedSearch } =
    useKibana().services;

  const { dataView } = useCreateDataView({ indexPatternString });

  return (
        <QueryStringInput
            appName="Observability"
            bubbleSubmitEvent={false}
            dataTestSubj={dataTestSubj}
            deps={{
              data,
              dataViews,
              docLinks,
              http,
              notifications,
              storage,
              uiSettings,
              unifiedSearch,
            }}
            disableAutoFocus
            disableLanguageSwitcher
            indexPatterns={dataView ? [dataView] : []}
            isDisabled={!indexPatternString}
            languageSwitcherPopoverAnchorPosition="rightDown"
            placeholder={placeholder}
            query={{ query, language: 'kuery' }}
            size="s"
            onChange={onChange}
        />
  );
}

```

Not saying you should use this, just sharing this implementation so you know it's also here.

Reply:
Very nice! I didn't even look past the the `MetricsExplorerKueryBar` since it was already used on the page. BUT I'm working on a prototype for something the the observability folder and I'm 100% going to use that bit there. I really like the `useCreateDataView` hook! That's awesome!